### DISCUSSION TITLE
Decoupled Navier-Stokes and energy equations using multiapp

### DISCUSSION THREAD

**Discussion by am-tc01** [idaholab/moose#20887] [Q&A General] [answered]
Hi,
I am trying to solve steady state Navier Stokes and energy equations in a decoupled manner using multiapp such that the main app solves for the temperature (T) and the sub-app solves for the velocity (v) and pressure (P). The driving force in my case is temperature (like in welding or AM) so the two systems are tightly coupled i.e. the temperature distribution depends on the flow field, while the flow field is driven by the gradient in temperature (Marangoni effect). My question now is, given that it is a steady state problem, do I need one sub-app like I described above or two sub-apps in the following way:
1. Main app solves for T without v, transfers T to sub-app
2. Sub-app solves for v and P with T as known, transfers v to sub-sub-app
3. Sub-sub-app solves for T with v as known

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You dont need to have 2 subapps or 3 apps in total. You just need to iterate this:
- Main app solves for T with v (starts with initial guess), transfers T to sub-app
- Sub-app solves for v and P with T as known, transfers v to main app

Even if the driving force is T, you can solve for velocity as the main app and solve for T in the subapp btw. The decision on subapp/main app depends on which app needs to take the smallest time steps. 

Guillaume

**Reply by am-tc01** [reply]
Hi Guillaume,

Thanks! Given the fact that it's a steady state problem, would it still work with 1 subapp? Do I need Fixed Point iteration in that case?

**Reply by GiudGiud** [reply]
You still need fixed point in that problem. Fluid flow is often easier to solve with relaxation-to-steady-state transients. DId you get a solution for steady state already?

**Reply by am-tc01** [reply]
Yes I did get a solution with steady state but the convergence for the fluid flow app is really slow.
I am not sure if I understood what you meant by "relaxation-to-steady-state transients"? Solve the steady state fluid flow as a transient problem with steady_state_detection, or?

BR.

**Reply by GiudGiud** [reply]
Yes that's the idea. The flow can reach the steady solution through a transient. 
This generally works better than trying to solve the steady problem directly.

**Reply by am-tc01** [reply]
I see. But I assume it would be slower than solving for steady state directly, given that one can actually get a solution with steady state, right? I should try it anyway. Thanks for the tip!

**Reply by GiudGiud** [reply]
Yes it's slower, but it ll work more reliably.
good luck, let us know how the coupling goes!
